model_compression_toolkit/core/common/graph/memory_graph/compute_graph_max_cut.py

```diff
@@ -47,9 +48,25 @@ def compute_graph_max_cut(memory_graph: MemoryGraph,
     l_bound = memory_graph.memory_lbound_single_op
     u_bound = 2 * sum([t.total_size for t in memory_graph.b_nodes]) - l_bound
     it = 0
+
+    @timeout_decorator.timeout(300)
```

Comment:
add a comment or even a logger info message explaining this restriction.
also, check if there is an elegant solution that doesn't use external pakages. It doesn't seems to save code or be more readable so maybe just use built-in time() check?